### ISSUE TITLE
SC2124 incorrectly triggered on array indexing

### ISSUE THREAD

**Issue by cdown** [koalaman/shellcheck#1141]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2124
- My shellcheck version (`shellcheck --version` or "online"): 0.4.7
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### Here's a snippet or screenshot that shows the problem:

`${@: -1}` here is a specific index, not an array:

```sh
$ cat script
#!/bin/bash

foo="${@: -1}"
echo "$foo"
$ ./script
./script
$ shellcheck script

In script line 3:
foo="${@: -1}"
    ^-- SC2124: Assigning an array to a string! Assign as array, or use * instead of @ to concatenate.
```

Found when reviewing https://github.com/cdown/clipmenu/pull/68.

**Comment by cdown** [comment]
I kind of get that this behaviour is derived from cases where `-2`/`-3`/etc are used, but in this case it's impossible for `-1` to reference more than one element

**Comment by koalaman** [comment]
This warning is correct for `${array[@]: -1}` because it expands to 0 or 1 elements, but interestingly `"${@: -1}"` always expands to exactly 1.
